--- origsrc/wildmidi-0.2.3.5/src/wildmidi.c	2013-09-15 14:36:17.990990700 -0500
+++ src/wildmidi-0.2.3.5/src/wildmidi.c	2013-09-15 15:30:41.622659800 -0500
@@ -291,7 +291,7 @@ close_wav_output ( void ) {
 
 HWAVEOUT hWaveOut;
 WAVEHDR header;
-unsigned long int mm_buffer_count;
+ULONG mm_buffer_count;
 static CRITICAL_SECTION waveCriticalSection;
 
 static int write_mm_output (char * output_data, int output_size);
@@ -301,11 +301,11 @@ WAVEHDR *mm_blocks;
 #define MM_BLOCK_SIZE 16384
 #define MM_BLOCK_COUNT 3
 
-unsigned long int mm_free_blocks = MM_BLOCK_COUNT;
-unsigned long int mm_current_block = 0;
+ULONG mm_free_blocks = MM_BLOCK_COUNT;
+ULONG mm_current_block = 0;
 
-static void CALLBACK mmOutProc( HWAVEOUT hWaveOut, UINT uMsg, DWORD dwInstance, DWORD dwParam1, DWORD dwParam2 ) {
-	int* freeBlockCounter = (int*)dwInstance;
+static void CALLBACK mmOutProc( HWAVEOUT hWaveOut, UINT uMsg, DWORD_PTR dwInstance, DWORD_PTR dwParam1, DWORD_PTR dwParam2 ) {
+	INT_PTR* freeBlockCounter = (INT_PTR*)dwInstance;
 	HWAVEOUT tmp_hWaveOut = hWaveOut;
 	DWORD tmp_dwParam1 = dwParam1;
 	DWORD tmp_dwParam2 = dwParam2;
