--- origsrc/wildmidi-0.2.3.4/configure.ac	2010-07-29 23:36:06.000000000 -0500
+++ src/wildmidi-0.2.3.4/configure.ac	2011-08-10 14:18:02.730615300 -0500
@@ -253,7 +253,7 @@ case "$arch" in
 esac
 
 case "$host_os" in
-        *mingw*)
+        cygwin*|*mingw*)
                 AUDDR="-lwinmm"
         ;;
         *)
--- origsrc/wildmidi-0.2.3.4/src/wildmidi.c	2010-07-29 19:33:46.000000000 -0500
+++ src/wildmidi-0.2.3.4/src/wildmidi.c	2011-08-10 14:26:06.309274400 -0500
@@ -37,13 +37,14 @@
 #include <termios.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
+#else
+#include <conio.h>
 #endif
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
 
 #if (defined _WIN32) || (defined __CYGWIN__)
-# include <conio.h>
 # include <windows.h>
 # include <mmsystem.h>
 #else
@@ -211,7 +212,7 @@ open_wav_output ( void ) {
 
 	if (wav_file[0] == '\0')
 		return -1;
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__CYGWIN__)
 	if ((audio_fd = open(wav_file, (O_RDWR | O_CREAT | O_TRUNC | O_BINARY))) < 0) {
 #else
 	if ((audio_fd = open(wav_file, (O_RDWR | O_CREAT | O_TRUNC), (S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH))) < 0) {
@@ -779,7 +780,7 @@ main (int argc, char **argv) {
 	unsigned long int seek_to_sample = 0;
     int inpause = 0;
 
-#ifndef _WIN32
+#if !defined(_WIN32) || defined(__CYGWIN__)
 	int my_tty;
 	struct termios _tty;
 	tcflag_t _res_oflg = _tty.c_oflag;
@@ -889,7 +890,7 @@ main (int argc, char **argv) {
 			WildMidi_Shutdown();
 			return 0;
 		}
-#ifndef _WIN32
+#if !defined(_WIN32) || defined(__CYGWIN__)
 		my_tty = fileno(stdin);
 		if (isatty(my_tty)) {
 			savetty();
@@ -950,7 +951,7 @@ main (int argc, char **argv) {
 					break;
 
 				ch = 0;
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__CYGWIN__)
 				if (_kbhit()) {
 					ch = _getch();
 					putch(ch);
@@ -991,7 +992,7 @@ main (int argc, char **argv) {
 							WildMidi_Shutdown();
 							printf("Shutting down Sound System\r\n");
 							close_output();
-#ifndef _WIN32
+#if !defined(_WIN32) || defined(__CYGWIN__)
 							if (isatty(my_tty))
 								resetty();
 #endif
@@ -1058,7 +1059,7 @@ main (int argc, char **argv) {
                         apr_mins, apr_secs, modes, master_volume,
                         pro_mins, pro_secs, perc_play);
 
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__CYGWIN__)
                     Sleep(5);
 #else
                     usleep(5000);
@@ -1114,7 +1115,7 @@ NEXTMIDI:
 		memset(output_buffer, 0, 16384);
 		send_output (output_buffer, 16384);
 		send_output (output_buffer, 16384);
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__CYGWIN__)
 		Sleep(5);
 #else
 		usleep(5000);
@@ -1124,7 +1125,7 @@ NEXTMIDI:
 			printf("oops\r\n");
 		printf("Shutting down Sound System\r\n");
 		close_output();
-#ifndef _WIN32
+#if !defined(_WIN32) || defined(__CYGWIN__)
 		if (isatty(my_tty))
 			resetty();
 #endif
