--- origsrc/wildmidi-0.2.2/configure.ac	2004-01-28 12:25:44.000000000 -0600
+++ src/wildmidi-0.2.2/configure.ac	2007-07-15 01:04:04.328125000 -0500
@@ -244,7 +244,7 @@
 esac
 
 case "$host_os" in
-        *mingw*)
+        cygwin*|*mingw*)
                 AUDDR="-lwinmm"
         ;;
         *)
--- origsrc/wildmidi-0.2.2/src/wildmidi.c	2004-01-28 12:13:46.000000000 -0600
+++ src/wildmidi-0.2.2/src/wildmidi.c	2007-07-15 00:59:18.953125000 -0500
@@ -43,7 +43,9 @@
 #include <unistd.h>
 
 #if (defined _WIN32) || (defined __CYGWIN__)
+#ifndef __CYGWIN__
 # include <conio.h>
+#endif
 # include <windows.h>
 # include <mmsystem.h>
 #else
@@ -827,7 +829,7 @@
 	static char *spinner="|/-\\";
 	static int spinpoint = 0;
 
-#ifndef _WIN32
+#if !defined(_WIN32) || defined(__CYGWIN__)
 	int my_tty;
 	struct termios _tty;
 	tcflag_t _res_oflg = _tty.c_oflag; 
@@ -936,7 +938,7 @@
 			WildMidi_Shutdown();
 			return 0;
 		}
-#ifndef _WIN32		
+#if !defined(_WIN32) || defined(__CYGWIN__)
 		my_tty = fileno(stdin);
 		if (isatty(my_tty)) {
 			savetty();
@@ -995,7 +997,7 @@
 					break;
 
 				ch = 0;
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__CYGWIN__)
 				if (_kbhit()) {
 					ch = _getch();
 					putch(ch);
@@ -1028,7 +1030,7 @@
 							WildMidi_Shutdown();
 							printf("Shutting down Sound System\n");
 							close_output();
-#ifndef _WIN32
+#if !defined(_WIN32) || defined(__CYGWIN__)
 							if (isatty(my_tty))
 								resetty();
 #endif
@@ -1106,7 +1108,7 @@
 			printf("oops\n");
 		printf("Shutting down Sound System\n");
 		close_output();
-#ifndef _WIN32
+#if !defined(_WIN32) || defined(__CYGWIN__)
 		if (isatty(my_tty))
 			resetty();
 #endif
